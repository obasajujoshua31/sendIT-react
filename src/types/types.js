export const signUpUser = "SIGN_UP_USER";
export const signInUser = "SIGN_IN_USER";
export const createOrder = "CREATE_ORDER";
export const createOrderSuccess = "CREATE_ORDER_SUCCESS";
export const createOrderFailure = "CREATE_ORDER_FAILURE";
export const viewOrders = "VIEW_ORDERS";
export const viewOrder = "VIEW_ORDER";
export const updateParcel = "UPDATE_PARCELS";
export const updateLocation = "UPDATE_PARCEL_LOCATION";
export const cancelOrder = "CANCEL_ORDER";
export const updateParcelStatus = "UPDATE_PARCEL_STATUS";
export const signInSuccess = "SIGN_IN_SUCCESS";
export const signInFailure = "SIGN_IN_FAILURE";
export const signUpSuccess = "SIGN_UP_SUCCESS";
export const signUpFailure = "SIGN_UP_FAILURE";
export const isLoggedOut = "LOGGED_OUT";
export const isAdmin = "IS_ADMIN";
