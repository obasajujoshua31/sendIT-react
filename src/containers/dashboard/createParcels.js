import React, { Component } from "react";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";
import Spinner from "react-md-spinner";
import createParcel from "../../actions/parcels/createOrders";
import { signInFailure, signUpFailure } from "../../types/types";
import validateParcelCreateForm, {
  isValid
} from "../../validations/parcelValidator";
import loadUserParcels from "../../actions/parcels/getParcels";

export class CreateOrder extends Component {
  constructor(props) {
    super(props);

    this.state = {
      parcelName: "",
      destination: "",
      pickUpLocation: "",
      weight: "",
      weightMetric: "Kg",
      isDeactivated: false,
      formErrors: {
        parcelName: "",
        destination: "",
        pickUpLocation: "",
        weight: ""
      },
      isLoading: true,
      parcels: []
    };
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  componentDidMount = () => {
    this.props.loadParcels();
  };

  componentDidUpdate(prevProps) {
    if (prevProps !== this.props) {
      if (
        this.props.authStatus === signInFailure ||
        this.props.authStatus === signUpFailure
      ) {
        this.props.history.push("/login");
      }
      if (this.props.parcels) {
        this.setState({ isLoading: false });
      }
      if (this.props.createdStatus) {
        this.props.history.push("view-orders");
      }
    }
  }

  handleChange(e) {
    const { name, value } = e.target;
    this.setState({ [name]: value });
    this.setState({ formErrors: validateParcelCreateForm(e, this.state) });
    if (!isValid(this.state)) {
      this.setState({ isDeactivated: true });
    } else {
      this.setState({ isDeactivated: false });
    }
  }

  handleSubmit(e) {
    e.preventDefault();
    this.setState({ isLoading: true });
    if (!isValid(this.state)) {
      return this.setState({ isLoading: false });
    } else {
      this.props.createParcel(this.state);
    }
  }

  render() {
    if (this.state.isLoading) {
      return (
        <div className="container">
          <div className="table-container">
            <div className="container-not-found">
              <Spinner />
            </div>
          </div>
        </div>
      );
    }

    const {
      parcelName,
      weight,
      pickUpLocation,
      destination,
      formErrors,
      isDeactivated
    } = this.state;
    return (
      <React.Fragment>
        <div className="account-wrapper">
          <div className="form-wrapper create-wrapper">
            <form onSubmit={this.handleSubmit} noValidate>
              <h2>Create Parcel</h2>

              <div className="form-oneline">
                <div className="form-group">
                  <label htmlFor="parcelName">Parcel Name</label>
                  <input
                    className={formErrors.parcelName.length > 0 ? "error" : ""}
                    type="text"
                    placeholder="Parcel Name"
                    name="parcelName"
                    value={parcelName}
                    onChange={this.handleChange}
                    required
                  />
                  {formErrors.parcelName.length > 0 && (
                    <span className="errorMsg">{formErrors.parcelName}</span>
                  )}
                </div>
                <div className="form-group">
                  <label htmlFor="parcel Weight">Weight</label>
                  <input
                    className={formErrors.weight.length > 0 ? "error" : ""}
                    type="number"
                    placeholder="Weight"
                    name="weight"
                    value={weight}
                    onChange={this.handleChange}
                    required
                  />
                  {formErrors.weight.length > 0 && (
                    <span className="errorMsg">{formErrors.weight} </span>
                  )}
                </div>
              </div>
              <div className="form-group">
                <label htmlFor="weightMetric">Weight Metric</label>
                <select
                  value={this.state.weightMetric}
                  onChange={this.handleChange}
                  name="weightMetric"
                >
                  <option value="Kg">Kg</option>
                  <option value="g">g</option>
                </select>
              </div>

              <div className="form-group">
                <label htmlFor="pickUpLocation">Pick Up Location</label>
                <input
                  className={
                    formErrors.pickUpLocation.length > 0 ? "error" : ""
                  }
                  type="text"
                  placeholder="Your PickUp Location"
                  name="pickUpLocation"
                  value={pickUpLocation}
                  onChange={this.handleChange}
                  required
                />
                {formErrors.pickUpLocation.length > 0 && (
                  <span className="errorMsg">{formErrors.pickUpLocation}</span>
                )}
              </div>

              <div className="form-group">
                <label htmlFor="destination">Your Destination</label>
                <input
                  className={formErrors.destination.length > 0 ? "error" : ""}
                  type="text"
                  placeholder="Your Destination"
                  name="destination"
                  value={destination}
                  onChange={this.handleChange}
                  required
                />
                {formErrors.destination.length > 0 && (
                  <span className="errorMsg">{formErrors.destination}</span>
                )}
              </div>
              <button hidden={isDeactivated} type="submit">
                Send Parcel
              </button>
            </form>
          </div>
        </div>
      </React.Fragment>
    );
  }
}
export const mapDispatchToProps = dispatch => ({
  loadParcels: () => dispatch(loadUserParcels()),
  createParcel: parcel => dispatch(createParcel(parcel))
});

export const mapStateToProps = ({ users, parcels }) => ({
  authStatus: users.singInStatus,
  createdStatus: parcels.parcelStatus,
  parcels: parcels.parcels
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(withRouter(CreateOrder));
