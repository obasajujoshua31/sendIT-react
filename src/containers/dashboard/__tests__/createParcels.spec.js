import React from "react";
import { shallow, mount } from "enzyme";
import sinon from "sinon";
import { CreateOrder, mapDispatchToProps } from "../createParcels";
import { signInFailure, signInSuccess } from "../../../types/types";

const props = {
  signUp: jest.fn(),
  history: {
    push: jest.fn()
  },
  errorMsg: "",
  signUpStatus: "",
  authStatus: "",
  createdStatus: "",
  createParcel: jest.fn(),
  loadParcels: jest.fn()
};

const mockParcel = {
  parcelName: "Monkey",
  destination: "Gorilla",
  pickUpLocation: "monkeygorillacom",
  weight: 34,
  weightMetric: "Kg"
};

const mockStore = {
  users: {
    signUpStatus: "Success",
    errorMsg: null
  }
};

const mockSubmit = {
  preventDefault: jest.fn()
};

const validState = {
  formErrors: {
    parcelName: "",
    destination: "",
    pickUpLocation: "",
    weight: "",
    weightMetric: ""
  }
};

const wrapper = shallow(<CreateOrder {...props} />);

describe(" <CreateOrder> ", () => {
  it("should render CreateOrder Component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("componentDidMount should call loadparcels", () => {
    const componentSpy = jest.spyOn(wrapper.instance(), "componentDidMount");

    componentSpy();
    expect(wrapper.instance().props.loadParcels).toBeCalled();
  });

  it("should set the parcelName field to test", () => {
    wrapper.setState({ isLoading: false });
    wrapper
      .find("input")
      .at(0)
      .simulate("change", {
        target: {
          value: "test",
          name: "parcelName"
        }
      });
    expect(wrapper.instance().state.parcelName).toEqual("test");
  });

  it("handle change with valid input should set isDeactivated to false", () => {
    wrapper.setState({
      formErrors: validState.formErrors,
      parcelName: "kddkkd",
      weight: 4,
      weightMetric: "Kg",
      destination: "lldldd",
      pickUpLocation: "dldlldll"
    });

    wrapper
      .find("input")
      .at(0)
      .simulate("change", {
        target: {
          value: "dkkdkkd",
          name: "parcelName"
        }
      });
    // console.log(wrapper.instance().state)
    expect(wrapper.instance().state.isDeactivated).toEqual(false);
  });

  it("should call dispatch action", () => {
    const dispatchSpy = sinon.spy();
    const dispatchResult = mapDispatchToProps(dispatchSpy);
    dispatchResult.loadParcels();
    dispatchResult.createParcel(mockParcel);
    expect(dispatchSpy.callCount).toEqual(2);
  });

  it("should call history.push", () => {
    wrapper.setProps({
      authStatus: signInFailure
    });

    expect(wrapper.instance().props.history.push).toHaveBeenCalled();
  });

  it("should call submit", () => {
    let prevented = false;
    const formElement = wrapper.find("form");
    formElement.simulate("submit", {
      preventDefault: () => {
        prevented = true;
      }
    });
    expect(prevented).toEqual(true);
  });

  it("should re render when the form is invalid when the submit button is clicked", () => {
    const submitSpy = jest.spyOn(wrapper.instance(), "handleSubmit");
    wrapper.setState({
      parcelName: "",
      destination: ""
    });
    submitSpy(mockSubmit);
    expect(wrapper).toMatchSnapshot();
  });

  it("should re render when there is error in the form", () => {
    wrapper.setState({
      formErrors: {
        parcelName: "kdkdkddkkd",
        destination: "llld",
        pickUpLocation: "kdkdkkd",
        weight: "kdkdkkd"
      }
    });
    wrapper
      .find("input")
      .at(0)
      .simulate("change", {
        target: {
          value: "test",
          name: ""
        }
      });

    expect(wrapper.instance().state.isDeactivated).toEqual(true);
    expect(wrapper).toMatchSnapshot();
  });

  it("should call props.createParcel", () => {
    let prevented = false;
    wrapper.setState({
      formErrors: validState.formErrors,
      parcelName: mockParcel.parcelName,
      destination: mockParcel.destination,
      pickUpLocation: mockParcel.pickUpLocation,
      weight: mockParcel.weight,
      weightMetric: mockParcel.weightMetric
    });
    const formElement = wrapper.find("form");
    formElement.simulate("submit", {
      preventDefault: () => {
        prevented = true;
      }
    });
    expect(wrapper.instance().props.createParcel).toHaveBeenCalled();
  });

  it("should call history.push", () => {
    wrapper.setProps({
      createdStatus: true
    });

    expect(wrapper.instance().props.history.push).toHaveBeenCalled();
  });

  it("should render when there the user is authenticated", () => {
    wrapper.setProps({
      authStatus: signInSuccess
    });
    expect(wrapper).toMatchSnapshot();
  });
});
