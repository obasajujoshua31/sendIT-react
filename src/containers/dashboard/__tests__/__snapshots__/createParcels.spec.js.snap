// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` <CreateOrder>  should re render when the form is invalid when the submit button is clicked 1`] = `
<Fragment>
  <div
    className="account-wrapper"
  >
    <div
      className="form-wrapper create-wrapper"
    >
      <form
        noValidate={true}
        onSubmit={
          [MockFunction] {
            "calls": Array [
              Array [
                Object {
                  "preventDefault": [MockFunction] {
                    "calls": Array [
                      Array [],
                    ],
                    "results": Array [
                      Object {
                        "type": "return",
                        "value": undefined,
                      },
                    ],
                  },
                },
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": <Fragment>
                  <div
                    className="account-wrapper"
                  >
                    <div
                      className="form-wrapper create-wrapper"
                    >
                      <form
                        noValidate={true}
                        onSubmit={
                          [MockFunction] {
                            "calls": Array [
                              Array [
                                Object {
                                  "preventDefault": [MockFunction] {
                                    "calls": Array [
                                      Array [],
                                    ],
                                    "results": Array [
                                      Object {
                                        "type": "return",
                                        "value": undefined,
                                      },
                                    ],
                                  },
                                },
                              ],
                            ],
                            "results": [Circular],
                          }
                        }
                      >
                        <h2>
                          Create Parcel
                        </h2>
                        <div
                          className="form-oneline"
                        >
                          <div
                            className="form-group"
                          >
                            <label
                              htmlFor="parcelName"
                            >
                              Parcel Name
                            </label>
                            <input
                              className=""
                              name="parcelName"
                              onChange={[Function]}
                              placeholder="Parcel Name"
                              required={true}
                              type="text"
                              value=""
                            />
                          </div>
                          <div
                            className="form-group"
                          >
                            <label
                              htmlFor="parcel Weight"
                            >
                              Weight
                            </label>
                            <input
                              className=""
                              name="weight"
                              onChange={[Function]}
                              placeholder="Weight"
                              required={true}
                              type="number"
                              value={4}
                            />
                          </div>
                        </div>
                        <div
                          className="form-group"
                        >
                          <label
                            htmlFor="weightMetric"
                          >
                            Weight Metric
                          </label>
                          <select
                            name="weightMetric"
                            onChange={[Function]}
                            value="Kg"
                          >
                            <option
                              value="Kg"
                            >
                              Kg
                            </option>
                            <option
                              value="g"
                            >
                              g
                            </option>
                          </select>
                        </div>
                        <div
                          className="form-group"
                        >
                          <label
                            htmlFor="pickUpLocation"
                          >
                            Pick Up Location
                          </label>
                          <input
                            className=""
                            name="pickUpLocation"
                            onChange={[Function]}
                            placeholder="Your PickUp Location"
                            required={true}
                            type="text"
                            value="dldlldll"
                          />
                        </div>
                        <div
                          className="form-group"
                        >
                          <label
                            htmlFor="destination"
                          >
                            Your Destination
                          </label>
                          <input
                            className=""
                            name="destination"
                            onChange={[Function]}
                            placeholder="Your Destination"
                            required={true}
                            type="text"
                            value=""
                          />
                        </div>
                        <button
                          hidden={false}
                          type="submit"
                        >
                          Send Parcel
                        </button>
                      </form>
                    </div>
                  </div>
                </Fragment>,
              },
            ],
          }
        }
      >
        <h2>
          Create Parcel
        </h2>
        <div
          className="form-oneline"
        >
          <div
            className="form-group"
          >
            <label
              htmlFor="parcelName"
            >
              Parcel Name
            </label>
            <input
              className=""
              name="parcelName"
              onChange={[Function]}
              placeholder="Parcel Name"
              required={true}
              type="text"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="parcel Weight"
            >
              Weight
            </label>
            <input
              className=""
              name="weight"
              onChange={[Function]}
              placeholder="Weight"
              required={true}
              type="number"
              value={4}
            />
          </div>
        </div>
        <div
          className="form-group"
        >
          <label
            htmlFor="weightMetric"
          >
            Weight Metric
          </label>
          <select
            name="weightMetric"
            onChange={[Function]}
            value="Kg"
          >
            <option
              value="Kg"
            >
              Kg
            </option>
            <option
              value="g"
            >
              g
            </option>
          </select>
        </div>
        <div
          className="form-group"
        >
          <label
            htmlFor="pickUpLocation"
          >
            Pick Up Location
          </label>
          <input
            className=""
            name="pickUpLocation"
            onChange={[Function]}
            placeholder="Your PickUp Location"
            required={true}
            type="text"
            value="dldlldll"
          />
        </div>
        <div
          className="form-group"
        >
          <label
            htmlFor="destination"
          >
            Your Destination
          </label>
          <input
            className=""
            name="destination"
            onChange={[Function]}
            placeholder="Your Destination"
            required={true}
            type="text"
            value=""
          />
        </div>
        <button
          hidden={false}
          type="submit"
        >
          Send Parcel
        </button>
      </form>
    </div>
  </div>
</Fragment>
`;

exports[` <CreateOrder>  should re render when there is error in the form 1`] = `
<Fragment>
  <div
    className="account-wrapper"
  >
    <div
      className="form-wrapper create-wrapper"
    >
      <form
        noValidate={true}
        onSubmit={
          [MockFunction] {
            "calls": Array [
              Array [
                Object {
                  "preventDefault": [MockFunction] {
                    "calls": Array [
                      Array [],
                    ],
                    "results": Array [
                      Object {
                        "type": "return",
                        "value": undefined,
                      },
                    ],
                  },
                },
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": <Fragment>
                  <div
                    className="account-wrapper"
                  >
                    <div
                      className="form-wrapper create-wrapper"
                    >
                      <form
                        noValidate={true}
                        onSubmit={
                          [MockFunction] {
                            "calls": Array [
                              Array [
                                Object {
                                  "preventDefault": [MockFunction] {
                                    "calls": Array [
                                      Array [],
                                    ],
                                    "results": Array [
                                      Object {
                                        "type": "return",
                                        "value": undefined,
                                      },
                                    ],
                                  },
                                },
                              ],
                            ],
                            "results": [Circular],
                          }
                        }
                      >
                        <h2>
                          Create Parcel
                        </h2>
                        <div
                          className="form-oneline"
                        >
                          <div
                            className="form-group"
                          >
                            <label
                              htmlFor="parcelName"
                            >
                              Parcel Name
                            </label>
                            <input
                              className="error"
                              name="parcelName"
                              onChange={[Function]}
                              placeholder="Parcel Name"
                              required={true}
                              type="text"
                              value=""
                            />
                            <span
                              className="errorMsg"
                            >
                              kdkdkddkkd
                            </span>
                          </div>
                          <div
                            className="form-group"
                          >
                            <label
                              htmlFor="parcel Weight"
                            >
                              Weight
                            </label>
                            <input
                              className="error"
                              name="weight"
                              onChange={[Function]}
                              placeholder="Weight"
                              required={true}
                              type="number"
                              value={4}
                            />
                            <span
                              className="errorMsg"
                            >
                              kdkdkkd
                               
                            </span>
                          </div>
                        </div>
                        <div
                          className="form-group"
                        >
                          <label
                            htmlFor="weightMetric"
                          >
                            Weight Metric
                          </label>
                          <select
                            name="weightMetric"
                            onChange={[Function]}
                            value="Kg"
                          >
                            <option
                              value="Kg"
                            >
                              Kg
                            </option>
                            <option
                              value="g"
                            >
                              g
                            </option>
                          </select>
                        </div>
                        <div
                          className="form-group"
                        >
                          <label
                            htmlFor="pickUpLocation"
                          >
                            Pick Up Location
                          </label>
                          <input
                            className="error"
                            name="pickUpLocation"
                            onChange={[Function]}
                            placeholder="Your PickUp Location"
                            required={true}
                            type="text"
                            value="dldlldll"
                          />
                          <span
                            className="errorMsg"
                          >
                            kdkdkkd
                          </span>
                        </div>
                        <div
                          className="form-group"
                        >
                          <label
                            htmlFor="destination"
                          >
                            Your Destination
                          </label>
                          <input
                            className="error"
                            name="destination"
                            onChange={[Function]}
                            placeholder="Your Destination"
                            required={true}
                            type="text"
                            value=""
                          />
                          <span
                            className="errorMsg"
                          >
                            llld
                          </span>
                        </div>
                        <button
                          hidden={true}
                          type="submit"
                        >
                          Send Parcel
                        </button>
                      </form>
                    </div>
                  </div>
                </Fragment>,
              },
            ],
          }
        }
      >
        <h2>
          Create Parcel
        </h2>
        <div
          className="form-oneline"
        >
          <div
            className="form-group"
          >
            <label
              htmlFor="parcelName"
            >
              Parcel Name
            </label>
            <input
              className="error"
              name="parcelName"
              onChange={[Function]}
              placeholder="Parcel Name"
              required={true}
              type="text"
              value=""
            />
            <span
              className="errorMsg"
            >
              kdkdkddkkd
            </span>
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="parcel Weight"
            >
              Weight
            </label>
            <input
              className="error"
              name="weight"
              onChange={[Function]}
              placeholder="Weight"
              required={true}
              type="number"
              value={4}
            />
            <span
              className="errorMsg"
            >
              kdkdkkd
               
            </span>
          </div>
        </div>
        <div
          className="form-group"
        >
          <label
            htmlFor="weightMetric"
          >
            Weight Metric
          </label>
          <select
            name="weightMetric"
            onChange={[Function]}
            value="Kg"
          >
            <option
              value="Kg"
            >
              Kg
            </option>
            <option
              value="g"
            >
              g
            </option>
          </select>
        </div>
        <div
          className="form-group"
        >
          <label
            htmlFor="pickUpLocation"
          >
            Pick Up Location
          </label>
          <input
            className="error"
            name="pickUpLocation"
            onChange={[Function]}
            placeholder="Your PickUp Location"
            required={true}
            type="text"
            value="dldlldll"
          />
          <span
            className="errorMsg"
          >
            kdkdkkd
          </span>
        </div>
        <div
          className="form-group"
        >
          <label
            htmlFor="destination"
          >
            Your Destination
          </label>
          <input
            className="error"
            name="destination"
            onChange={[Function]}
            placeholder="Your Destination"
            required={true}
            type="text"
            value=""
          />
          <span
            className="errorMsg"
          >
            llld
          </span>
        </div>
        <button
          hidden={true}
          type="submit"
        >
          Send Parcel
        </button>
      </form>
    </div>
  </div>
</Fragment>
`;

exports[` <CreateOrder>  should render CreateOrder Component 1`] = `
<div
  className="container"
>
  <div
    className="table-container"
  >
    <div
      className="container-not-found"
    >
      <MDSpinner
        color1="rgb(66, 165, 245)"
        color2="rgb(239, 83, 80)"
        color3="rgb(253, 216, 53)"
        color4="rgb(76, 175, 80)"
        duration={1333}
        size={28}
      />
    </div>
  </div>
</div>
`;

exports[` <CreateOrder>  should render when there the user is authenticated 1`] = `
<Fragment>
  <div
    className="account-wrapper"
  >
    <div
      className="form-wrapper create-wrapper"
    >
      <form
        noValidate={true}
        onSubmit={
          [MockFunction] {
            "calls": Array [
              Array [
                Object {
                  "preventDefault": [MockFunction] {
                    "calls": Array [
                      Array [],
                    ],
                    "results": Array [
                      Object {
                        "type": "return",
                        "value": undefined,
                      },
                    ],
                  },
                },
              ],
              Array [
                Object {
                  "preventDefault": [Function],
                },
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": <Fragment>
                  <div
                    className="account-wrapper"
                  >
                    <div
                      className="form-wrapper create-wrapper"
                    >
                      <form
                        noValidate={true}
                        onSubmit={
                          [MockFunction] {
                            "calls": Array [
                              Array [
                                Object {
                                  "preventDefault": [MockFunction] {
                                    "calls": Array [
                                      Array [],
                                    ],
                                    "results": Array [
                                      Object {
                                        "type": "return",
                                        "value": undefined,
                                      },
                                    ],
                                  },
                                },
                              ],
                              Array [
                                Object {
                                  "preventDefault": [Function],
                                },
                              ],
                            ],
                            "results": [Circular],
                          }
                        }
                      >
                        <h2>
                          Create Parcel
                        </h2>
                        <div
                          className="form-oneline"
                        >
                          <div
                            className="form-group"
                          >
                            <label
                              htmlFor="parcelName"
                            >
                              Parcel Name
                            </label>
                            <input
                              className=""
                              name="parcelName"
                              onChange={[Function]}
                              placeholder="Parcel Name"
                              required={true}
                              type="text"
                              value="Monkey"
                            />
                          </div>
                          <div
                            className="form-group"
                          >
                            <label
                              htmlFor="parcel Weight"
                            >
                              Weight
                            </label>
                            <input
                              className=""
                              name="weight"
                              onChange={[Function]}
                              placeholder="Weight"
                              required={true}
                              type="number"
                              value={34}
                            />
                          </div>
                        </div>
                        <div
                          className="form-group"
                        >
                          <label
                            htmlFor="weightMetric"
                          >
                            Weight Metric
                          </label>
                          <select
                            name="weightMetric"
                            onChange={[Function]}
                            value="Kg"
                          >
                            <option
                              value="Kg"
                            >
                              Kg
                            </option>
                            <option
                              value="g"
                            >
                              g
                            </option>
                          </select>
                        </div>
                        <div
                          className="form-group"
                        >
                          <label
                            htmlFor="pickUpLocation"
                          >
                            Pick Up Location
                          </label>
                          <input
                            className=""
                            name="pickUpLocation"
                            onChange={[Function]}
                            placeholder="Your PickUp Location"
                            required={true}
                            type="text"
                            value="monkeygorillacom"
                          />
                        </div>
                        <div
                          className="form-group"
                        >
                          <label
                            htmlFor="destination"
                          >
                            Your Destination
                          </label>
                          <input
                            className=""
                            name="destination"
                            onChange={[Function]}
                            placeholder="Your Destination"
                            required={true}
                            type="text"
                            value="Gorilla"
                          />
                        </div>
                        <button
                          hidden={true}
                          type="submit"
                        >
                          Send Parcel
                        </button>
                      </form>
                    </div>
                  </div>
                </Fragment>,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          }
        }
      >
        <h2>
          Create Parcel
        </h2>
        <div
          className="form-oneline"
        >
          <div
            className="form-group"
          >
            <label
              htmlFor="parcelName"
            >
              Parcel Name
            </label>
            <input
              className=""
              name="parcelName"
              onChange={[Function]}
              placeholder="Parcel Name"
              required={true}
              type="text"
              value="Monkey"
            />
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="parcel Weight"
            >
              Weight
            </label>
            <input
              className=""
              name="weight"
              onChange={[Function]}
              placeholder="Weight"
              required={true}
              type="number"
              value={34}
            />
          </div>
        </div>
        <div
          className="form-group"
        >
          <label
            htmlFor="weightMetric"
          >
            Weight Metric
          </label>
          <select
            name="weightMetric"
            onChange={[Function]}
            value="Kg"
          >
            <option
              value="Kg"
            >
              Kg
            </option>
            <option
              value="g"
            >
              g
            </option>
          </select>
        </div>
        <div
          className="form-group"
        >
          <label
            htmlFor="pickUpLocation"
          >
            Pick Up Location
          </label>
          <input
            className=""
            name="pickUpLocation"
            onChange={[Function]}
            placeholder="Your PickUp Location"
            required={true}
            type="text"
            value="monkeygorillacom"
          />
        </div>
        <div
          className="form-group"
        >
          <label
            htmlFor="destination"
          >
            Your Destination
          </label>
          <input
            className=""
            name="destination"
            onChange={[Function]}
            placeholder="Your Destination"
            required={true}
            type="text"
            value="Gorilla"
          />
        </div>
        <button
          hidden={true}
          type="submit"
        >
          Send Parcel
        </button>
      </form>
    </div>
  </div>
</Fragment>
`;
